tests/explicit_delete.cpp

```diff
@@ -0,0 +1,39 @@
+#include "lua_test.hpp"
+
+class Deletable : public luabind::Object {
+public:
+    Deletable() = default;
+    ~Deletable() override {
+        deletedCount++;
+    }
+
+public:
+    static int deletedCount;
+};
+
+int Deletable::deletedCount = 0;
+
+class ExplicitDeleteTest : public LuaTest {
+protected:
+    void SetUp() override {
+        luabind::class_<Deletable>(L, "Deletable").constructor<>("new").construct_shared<>("makeShared");
+
+        EXPECT_EQ(lua_gettop(L), 0);
+    }
+};
+
+TEST_F(ExplicitDeleteTest, ExplicitDelete) {
+    run(R"--(
+        obj = Deletable:new()
+        obj:delete()
```

Comment:
Make sure that double deletion, either manually or from gc will pose a problem.

Reply:
Sorry didn't comprehend what you mean. If we call delete() on the same object twice the behavior is defined, first delete will call the user_data destructor which will delete the data, the second will not call user_data destructor as object stored in user_data is already null.

Reply:
Sorry, meant to write "will not pose a problem". 
I thing this test should cover that:
``` lua
obj = Deletable:new()
obj:delete()
obj:delete()
obj = null
collectgarbage()
```

Reply:
Done.